api/operator/v1alpha1/features.go

```diff
@@ -0,0 +1,22 @@
+package v1alpha1
+
+import (
+	"k8s.io/component-base/featuregate"
+)
+
+var (
+	// TechPreview: v1.15
+	//
+	// IstioCSR enables the controller for istiocsr.operator.openshift.io resource,
+	// which extends cert-manager-operator to deploy and manage the istio-csr agent.
+	// OpenShift Service Mesh facilitates the integration and istio-csr is an agent that
+	// allows Istio workload and control plane components to be secured using cert-manager.
+	//
+	// For more details,
+	// https://github.com/openshift/enhancements/blob/master/enhancements/cert-manager/istio-csr-controller.md
+	FeatureIstioCSR featuregate.Feature = "IstioCSR"
+)
+
+var OperatorFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
+	FeatureIstioCSR: {Default: false, PreRelease: "TechPreview"},
```

Comment:
Is there a check specific to PreRelease field?

Reply:
Upstream uses GA, beta, alpha for this prerelease field values;
but avoided Beta/Alpha since we do not use them in our lifecycle.

Reply:
> Is there a check specific to PreRelease field?

Added a test case in features_test to ensure that `FeatureSpec{Prerelease: "TechPreview"}` features are always disabled by default.

Reply:
UX question: How would a user enable a supported addon (once goes GA), or is it enabled by default?

Reply:
Will be enabled by default when the feature moves to GA. We just need to update to `FeatureSpec{Default: true, Prerelease: "GA"}` and it should work.

Reply:
Well, wouldn't it be kind of expensive to enable that controller for users who don't need this addon at all? Not sure how much this would impact the overall performance. Perhaps we could experiment with it when going GA, at least it shouldn't be an issue at this time 😄